manifests/init.pp

```diff
@@ -137,7 +137,7 @@
   Optional[String] $archive_source,
   String $cfg_location,
   Variant[Hash,Sensitive[Hash]] $cfg,
-  Optional[Variant[Hash,Array]] $ldap_cfg,
+  Optional[Variant[Hash,Array[Hash],Sensitive[Hash],Sensitive[Array[Hash]]]] $ldap_cfg,
```

Comment:
now thats a combination that could be put into ita own type with some tests

Reply:
Maybe if we ever get round to converting the docs to puppet-strings??  Tests are mainly useful for Pattern types IMO.